src/contracts/core/Slasher.sol

```diff
@@ -27,7 +28,218 @@ import "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
+    ) external {
+        require(bipsToSlash <= SlashingAccountingUtils.BIPS_FACTOR, "invalid slashing amount");
+        require(bipsToSlash != 0, "cannot slash 0");
+        address avs = msg.sender;
+        int256 epoch = SlashingAccountingUtils.currentEpoch();
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            IStrategy strategy = strategies[i];
+            uint256 requestedSlashedBipsBefore = requestedSlashedBips[operator][strategy][epoch][avs];
+            uint256 requestedSlashedBipsAfter = requestedSlashedBipsBefore + bipsToSlash;
+            // TODO: consider loss of precision here. this is undesirable; we should fix it
+            uint256 bipsAllowedToSlash =
+                getSlashableBips(operator, avs, strategy, epoch) * getMaxSlashingRate(avs, strategy, epoch) / SlashingAccountingUtils.BIPS_FACTOR;
+            int256 changeInPendingSlashedBips = _changeInPendingSlashedBips({
+                bipsAllowedToSlash: bipsAllowedToSlash,
+                requestedSlashedBipsBefore: requestedSlashedBipsBefore,
+                requestedSlashedBipsAfter: requestedSlashedBipsAfter
+            });
+
+            // TODO: events
+            requestedSlashedBips[operator][strategy][epoch][avs] = requestedSlashedBipsAfter;
+            pendingSlashedBips[operator][strategy][epoch] = uint256(
+                int256(pendingSlashedBips[operator][strategy][epoch]) + int256(changeInPendingSlashedBips)
+            );
+        }
+    }
+
+    function reduceRequestedSlashedBips(
+        address operator,
+        IStrategy[] memory strategies,
+        // TODO: naming
+        uint256 bipsToReduce
+    ) external {
+        require(bipsToReduce != 0, "cannot reduce slashing by 0");
+        address avs = msg.sender;
+        int256 epoch = SlashingAccountingUtils.currentEpoch();
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            IStrategy strategy = strategies[i];
+            uint256 requestedSlashedBipsBefore = requestedSlashedBips[operator][strategy][epoch][avs];
+            uint256 requestedSlashedBipsAfter;
+            // ignore underflow; max out at reducing to zero
+            if (bipsToReduce >= requestedSlashedBipsBefore) {
+                requestedSlashedBipsAfter = 0;
+            } else {
+                requestedSlashedBipsAfter = requestedSlashedBipsBefore - bipsToReduce;
+            }
+            // TODO: consider loss of precision here. this is undesirable; we should fix it
+            uint256 bipsAllowedToSlash =
+                getSlashableBips(operator, avs, strategy, epoch) * getMaxSlashingRate(avs, strategy, epoch) / SlashingAccountingUtils.BIPS_FACTOR;
+            int256 changeInPendingSlashedBips = _changeInPendingSlashedBips({
+                bipsAllowedToSlash: bipsAllowedToSlash,
+                requestedSlashedBipsBefore: requestedSlashedBipsBefore,
+                requestedSlashedBipsAfter: requestedSlashedBipsAfter
+            });
```

Comment:
this implementation multiplies slashable bips by max slashing rate and then compares that to the requested slashed bips, but what it should be doing is comparing the requested slashed bips to the max slashing rate and then multiplying by the slashable bips

Reply:
h/t to @8sunyuan for pointing this out

Reply:
I believe I've addressed this in https://github.com/Layr-Labs/eigenlayer-contracts/pull/543/commits/8a9447a260f59fe2d3a94c210b68067592bcdced